modules/core/src/weaver/suites.scala

```diff
@@ -92,19 +95,28 @@ abstract class MutableFSuite[F[_]] extends RunnableSuite[F]  {
     def usingRes(run : Res => F[Expectations]) : Unit = apply(run)
   }
 
+  def isCI: Boolean = "true" == System.getenv("CI")
+
   override def spec(args: List[String]) : Stream[F, TestOutcome] =
     synchronized {
       if (!isInitialized) isInitialized = true
-      val testsNotIgnored = testSeq.filterNot(_._1.tags(TestName.Tags.ignore))
-      val testsTaggedOnly = testsNotIgnored.filter(_._1.tags(TestName.Tags.only))
-      val filteredTests = if (testsTaggedOnly.isEmpty) {
+      val testsNotIgnored: Seq[(TestName, Res => F[TestOutcome])] = testSeq.filterNot(_._1.tags(TestName.Tags.ignore))
+      val testsTaggedOnly: Seq[(TestName, Res => F[TestOutcome])]  = testSeq.filter(_._1.tags(TestName.Tags.only))
+      val onlyTestsNotIgnored = testsTaggedOnly.filter(taggedOnly => testsNotIgnored.contains(taggedOnly))
+      val filteredTests = if (onlyTestsNotIgnored.isEmpty) {
         val argsFilter = Filters.filterTests(this.name)(args)
         testsNotIgnored.collect {
           case (name, test) if argsFilter(name) => test
         }
-      } else testsTaggedOnly.map(_._2)
+      } else onlyTestsNotIgnored.map(_._2)
       val parallism = math.max(1, maxParallelism)
       if (filteredTests.isEmpty) Stream.empty // no need to allocate resources
+      else if (testsTaggedOnly.nonEmpty && isCI) {
+        val failureOutcomes = testsTaggedOnly
+          .map(_._1)
+          .map(onlyNotOnCiFailure)
+        Stream.emits(failureOutcomes).lift[F](effectCompat.effect)
+      }
```

Comment:
Do we want the tests to fail if there would end up being no tests to run, but there was something tagged `.only`? For example, a suite with just one test that has both `.only` and `.ignore` would be caught in the first if condition here and so we would return a `Stream.empty`, i.e., no tests would be run at all (and the `.only` is ignored).

On the other hand, if there are _any_ tests to be run, and any of the other tests are marked as `.only` and `.ignore` we will fail (we will report all tests marked with `.only`)

Reply:
I think I'd swap both conditions so that the `isCI` flag would be checked first, and result in a failure if enabled and in the presence of any `.only`. The reason is that the behaviour would be surprising in a later PR when tests are added and there might be a cognitive disconnect between the suite failing and the addition of `.only`.

Reply:
done